component/component/main.jsonnet

```diff
@@ -94,10 +100,28 @@ local secrets = [
                       valueFrom: {
                         secretKeyRef: {
                           key: 'api_secret',
-                          name: 'exoscale',
+                          name: cronjob_secret_name,
                         },
                       },
                     },
+                    {
+                      name: 'K8S_SERVER_URL',
+                      valueFrom: {
+                        secretKeyRef: {
+                          key: 'cluster_server',
+                          name: cronjob_secret_name,
+                        },
+                      },
+                    },
+                     {
+                       name: 'K8S_TOKEN',
+                       valueFrom: {
+                         secretKeyRef: {
+                           key: 'cluster_token',
+                           name: cronjob_secret_name,
+                         },
+                       },
+                     },
```

Comment:
Combined with the suggestion above, we can simplify this to:
```yaml
envFrom:
  - secretRef:
      name: cronjob_secret_name
```
that way you don't need to specify every env var individually, making it more maintainable here.